### PR TITLE
Add `GPTNeoXPreprocessor`

### PR DESCRIPTION
**Pull request by shivance**
In favor of #1052

### CODE REVIEW

**Comment by shivance** [comment]
This PR is still a work in progress.

**Comment by shivance** [comment]
@mattdangerw should we pull in Presets too, through this PR or do a seperate PR for that ?

**Comment by mattdangerw** [comment]
> @mattdangerw should we pull in Presets too, through this PR or do a seperate PR for that ?

Let's do separate! Smaller is better!

**Comment by shivance** [comment]
Hey @mattdangerw this PR is ready for review !

File: keras_nlp/models/gpt_neo_x/gpt_neo_x_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        x: A string, `tf.Tensor` or list of python strings.
+        y: Any label data. Will be passed through unaltered.
+        sample_weight: Any label weight data. Will be passed through unaltered.
+        sequence_length: Pass to override the configured `sequence_length` of
+            the layer.
+
+    Examples:
+
+    Directly calling the layer on data.
+    ```python
```
Comment (mattdangerw) [inline keras_nlp/models/gpt_neo_x/gpt_neo_x_preprocessor.py:73]: Let's leave the code examples out for now, as they rely on `from_preset` and won't quite work yet. Like we did with Tokenizer. We can add them later (and the preset names will be a bit different probably, all lowercase.

**Review by mattdangerw — APPROVED** [review]
This looks great! Just one comment

**Comment by shivance** [comment]
Sounds good @mattdangerw ! 
Made the change

**Pull request merged**